improver/calibration/reliability_calibration.py

```diff
@@ -1324,3 +1327,139 @@ def process(self, forecast: Cube, reliability_table: Union[Cube, CubeList]) -> C
             warnings.warn(msg)
 
         return calibrated_forecast
+
+    def _point_by_point_calibration(
+        self, forecast: Cube, reliability_table: CubeList,
+    ) -> Cube:
+        """
+        Apply point by point reliability calibration by iteratively picking a spatial
+        coordinate within the forecast cube, extracting the forecast at that point
+        and the reliability table corresponding to that point, then passing the
+        extracted forecast and reliability table to _get_calibrated_forecast().
+
+        Args:
+            forecast:
+                The forecast to be calibrated.
+            reliability_table:
+                The reliability table to use for applying calibration.
+
+        Returns:
+            The forecast cube following calibration.
+        """
+
+        calibrated_cubes = iris.cube.CubeList()
+        y_name = forecast.coord(axis="y").name()
+        x_name = forecast.coord(axis="x").name()
+
+        # create list of dimensions other than time dimension
+        # and threshold dimension
+        dim_names = get_dim_coord_names(forecast)
+        dim_associated_coords = []
+
+        # create lists with first entry a dimension name and second
+        # entry a list of auxiliary coordinates associated with that
+        # dimension. Collect these lists together into a list.
+        for dim_name in dim_names:
+            dim = forecast.coord_dims(dim_name)
+            template_dim = [x for x in dim]
+            associated_coords = [
+                c
+                for d in template_dim
+                for c in forecast.coords(dimensions=d, dim_coords=False)
+            ]
+            dim_associated_coords.append([dim_name, associated_coords])
```

Comment:
Related to [this comment](https://github.com/metoppv/mo-blue-team/issues/404#issuecomment-1311963471), we probably want to think about the auxiliary coordinate handling and the use case for this being added.